### PR TITLE
Add the first version of Repository Management

### LINKED ISSUE #4664: Gutenberg repository management document, v1
To better communicate how things are done around this GitHub repo, I've drafted a [Gutenberg repository management document](https://paper.dropbox.com/doc/Gutenberg-Repository-Management-dnkv20djioiT9BSbNpGQl). Specifically, it covers:

* Issues
    * Labels
    * Triaging Issues
* Pull Requests
    * Code Review
    * Merging Pull Requests
    * Closing Pull Requests

Please leave comments on the document as you see fit. Generally, it's a consolidated version of WP-CLI's [pull requests](https://make.wordpress.org/cli/handbook/pull-requests/), [code review](https://make.wordpress.org/cli/handbook/code-review/), and [repository management](https://make.wordpress.org/cli/handbook/repository-management/) documents. We can add further content to it as we go along.

Once we agree it's in a good place, I'll submit it as a pull request against this repo. For sign-off, I'm looking for a "This looks fine to me". Thanks!

### PR DESCRIPTION
**Pull request by danielbachhuber**
Meant to be a first version we can iterate on, not the final version.

Fixes #4664

### CODE REVIEW

**Comment by gziolo** [comment]
Thanks for starting this PR. The part related to code, PRs and review process looks solid 👍

File: docs/manifest.json
Comment (jeffpaul) [inline docs/manifest.json]: Won't this slug conflict with the `Release Process` (release.md) one above?

Comment (danielbachhuber) [reply docs/manifest.json]: Good catch. Updated in 87c867880b363bb5f08b8a24f76c3a8d63e01a34

File: docs/repository-management.md
Comment (jeffpaul) [inline docs/repository-management.md]: @karmatosed this first bullet is confusing, particularly after the comma. Would you please try to re-state your intent for this bullet?

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: These links don't work because the Markdown bolding `**` don't assign `id`. Might want to use a nested heading level instead, i.e. `###`

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: Another descriptor for healthy issue backlog:

- Specific: No broad mega-issues, because they're difficult to close. Ideally an issue can be closed in a single pull request.

Comment (aduth) [reply docs/repository-management.md]: Maybe under "Actionable", making clear the steps to reproduce a particular issue. It's difficult and time-consuming to debug an issue when it's specific to a set of steps.

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: I'd need to double-check, but I'm unsure if labels can be assigned if not being a member of the project.

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: Should this be `[Status]`-prefixed?

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: There's now also a "Stale" label, to help identify issues which might need to be closed. Wondering if this is redundant with "Needs More Info" though; the latter might not be as great a fit for pull requests.

cc @youknowriad (I believe you added the label?)

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: This link is access-restricted (non-members cannot view).

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: Can have some complexity for forked repositories (need to add as a remote and fetch).

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: While it sounds nice, I'm not sure how one is expected to interpret this 😄

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: Maybe we could add some bits about:

- Anyone can do a code review, even newcomers.
- GitHub's code review workflows ("Request Changes" vs. "Approve" vs. "Comment")

File: docs/repository-management.md
Comment (aduth) [inline docs/repository-management.md]: Noting that if you intend for this quote to be separated new lines, you need a blank line between each. Otherwise it's rendered as a single line.

```
> Thanks ________ for the time you’ve spent on this pull request.
>
> I’m closing this pull request because ________. To clarify further, ________.
>
> For more details, please see ________ and ____.
```

**Review by aduth — CHANGES_REQUESTED** [review]

**Comment by aduth** [comment]
Has it been considered to add a reference to this document from the base `CONTRIBUTING.md` file?

File: docs/repository-management.md
Comment (danielbachhuber) [reply docs/repository-management.md]: Good catch cae7bd474d33319eecec5b8396e6c84e793f5f1a

File: docs/repository-management.md
Comment (danielbachhuber) [reply docs/repository-management.md]: Potentially. In the Dropbox document, we were discussing whether it was a label we wanted to have. If we want to keep it, we can prefix with `[Status]`. Otherwise, we'll need to work out an equivalent.

File: docs/repository-management.md
Comment (danielbachhuber) [reply docs/repository-management.md]: Ideally, we'd make the team public so it can be `@mentioned` as needed.

File: docs/repository-management.md
Comment (danielbachhuber) [reply docs/repository-management.md]: Fixed in 3f8cf190

File: docs/repository-management.md
Comment (karmatosed) [reply docs/repository-management.md]: I think we should remove 'Product Decision' until we as a group agree on it's use - we haven't yet. I am fine if we do but just now going through the label, it was being wildly used differently from this. I stand by having a label like this encourages it's overuse and wrong use. For now please remove. Then, let's as a team discuss this and once a decision has been made it can be iterated on by adding back in. We can of course do this if we agree but just a triage of that issue had vastly different interpretations, for now I have removed it's use. I will add back in if we decide - nothing is unable to be reversed.

**Review by karmatosed — CHANGES_REQUESTED** [review]

File: docs/repository-management.md
Comment (karmatosed) [reply docs/repository-management.md]: It maybe is a matter of context but 'needs more info' to me means literally that. For example, what browser or device someone was using.

File: docs/repository-management.md
Comment (karmatosed) [reply docs/repository-management.md]: +1 for showing GitHub's review workflow here and also using as a call for encouraging new reviewers.

File: docs/repository-management.md
Comment (aduth) [reply docs/repository-management.md]: >Ideally, we'd make the team public so it can be `@mentioned` as needed.

Is this possible in GitHub? They make a distinction between "Secret" and "Visible", but even for "Visible" teams you have to be a member of the organization. Which is to say: "Gutenberg Core" is a Visible team, yet is still access-restricted to list users to those within the organization.

Comment (danielbachhuber) [reply docs/repository-management.md]: Good point — I don't know for certainty.

File: docs/repository-management.md
Comment (danielbachhuber) [reply docs/repository-management.md]: Great clarification, @karmatosed. I've removed the label from the list for now in a4a46f60

File: docs/repository-management.md
Comment (jeffpaul) [reply docs/repository-management.md]: Bah, looks like only org members can see teams so probably best to remove link. https://help.github.com/articles/changing-team-visibility/

Comment (danielbachhuber) [reply docs/repository-management.md]: The reason I mention is that you can assign an entire team for pull request review:

image

If we want to institute a workflow where core team members have final say on pull request merge, using this GitHub feature seems like a useful way to make the final request.

File: docs/repository-management.md
Comment (gziolo) [reply docs/repository-management.md]: I added the `Stale` label to simplify triaging. It's difficult to find out which PRs were already reviewed otherwise. I think about adding another label to mark that PR was triaged and needs a review to proceed. I'm not fully sold on the latter though.

File: docs/repository-management.md
Comment (jaswrks) [inline docs/repository-management.md]: Everything after the comma in this line could probably go. It seems confusing.

Love this document! Such great clarity in this 💯

**Comment by danielbachhuber** [comment]
@karmatosed @aduth What remaining changes would you like to see before we land this? If it's easier, can you simply make the changes directly? Thanks

File: docs/repository-management.md
Comment (aduth) [reply docs/repository-management.md]: Maybe drop the link but make mention of `WordPress/gutenberg-core` team as having final say? Or simply list the members of the team somewhere?

**Review by aduth — APPROVED** [review]
A few items of feedback remain unaddressed and relevant, but this is a good first pass.

File: docs/repository-management.md
Comment (danielbachhuber) [reply docs/repository-management.md]: Sounds good. I've changed to:

> The final pull request merge decision is made by the **@wordpress/gutenberg-core** team.

2d856c2b

**Comment by danielbachhuber** [comment]
> Has it been considered to add a reference to this document from the base `CONTRIBUTING.md` file?

Added in 1f47b618

File: CONTRIBUTING.md
Comment (aaronjorbin) [inline CONTRIBUTING.md]: The tone on this seems a bit offputting.  Maybe something like,

`The workflow is documented in greater detail in the [repository_management] document.`

File: docs/repository-management.md
Comment (aaronjorbin) [inline docs/repository-management.md]: Maybe add: "If there is not enough information to confirm the report, add the `Needs More Info` status.

File: CONTRIBUTING.md
Comment (danielbachhuber) [reply CONTRIBUTING.md]: Good point. Updated in c0e24ab3

File: docs/repository-management.md
Comment (danielbachhuber) [reply docs/repository-management.md]: Updated in c2709532

**Review by aaronjorbin — APPROVED** [review]

**Comment by danielbachhuber** [comment]
@WordPress/gutenberg-core What's remaining in order for this to land?

**Comment by youknowriad** [comment]
Most of the feedback looks addressed and you have two approvals. This is good to merge for me. Let's iterate on it.

**Pull request merged**
